protocol-units/settlement/mcr/contracts/script/MovementDeployer.s.sol

```diff
@@ -0,0 +1,269 @@
+pragma solidity ^0.8.13;
+
+import "forge-std/Script.sol";
+import {MOVEToken} from "../src/token/MOVEToken.sol";
+import {stlMoveToken} from "../src/token/stlMoveToken.sol";
+import {MovementStaking} from "../src/staking/MovementStaking.sol";
+import {MCR} from "../src/settlement/MCR.sol";
+import {ProxyAdmin} from "@openzeppelin/contracts/proxy/transparent/ProxyAdmin.sol";
+import {TransparentUpgradeableProxy} from "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+import {TimelockController} from "@openzeppelin/contracts/governance/TimelockController.sol";
+import { Helper } from "./helpers/Helper.sol";
+import { Vm } from "forge-std/Vm.sol";
+import { stdJson } from "forge-std/StdJson.sol";
+
+contract MovementDeployer is Script {
+
+    using stdJson for string;
+
+    TransparentUpgradeableProxy public mcrProxy;
+    TransparentUpgradeableProxy public stakingProxy;
+    TransparentUpgradeableProxy public stlMoveProxy;
+    TransparentUpgradeableProxy public moveProxy;
+    ProxyAdmin public admin;
+    TimelockController public timelock;
+    Deployment public deployment;
+    Helper helper = new Helper();
+    bool public record = true;
+
+    string public mcrSignature = "initialize(address,uint256,uint256,uint256,address[])";
+    string public stakingSignature = "initialize(address)";
+    string public stlMoveSignature = "initialize(string,string,address)";
+    string public moveSignature = "initialize()";
+    address public ZERO = address(0x0);
+
+    struct Deployment {
+        address mcr;
+        address mcrAdmin;
+        address staking;
+        address stakingAdmin;
+        address move;
+        address moveAdmin;
+        address stlMove;
+        address stlMoveAdmin;
+        address timelock;
+        address multisig;
+    }
+
+    function run() external {
+        uint256 minDelay = 1 days;
+        address[] memory proposers = new address[](5);
+        address[] memory executors = new address[](1);
+        proposers[0] = helper.s2a("Andy");
+        proposers[1] = helper.s2a("Bob");
+        proposers[2] = helper.s2a("Charlie");
+        proposers[3] = helper.s2a("David");
+        proposers[4] = helper.s2a("Eve");
+        executors[0] = helper.s2a("MultisigAddress");
+        address adminAddress = ZERO;
+
+        _loadDeployments();
+
+        uint256 signer = vm.envUint("PRIVATE_KEY");
```

Comment:
Ideally, we won't be loading cryptographic material from the environment even when said environment is a guarded CI instance. We can address this is a future PR though.